### PR TITLE
Au revoir, GLX

### PR DESCRIPTION
**Pull request by GeorgesStavracas**
### Description

Drop all GLX-specific code, and rely only on EGL.

### Motivation and Context

GLX is legacy, and has various quirks and complications that make supporting it a chore. Furthermore, GLX doesn't easily support modern texture-sharing mechanisms (DMA-BUF), which complicates it even more.

Since our EGL code paths have matured over the years, it is time to consider dropping GLX.

### How Has This Been Tested?

With this branch, try and perform routine tasks using X11, and the xcomposite plugins. Everything should works at least equally, or better, than before.

### Types of changes
 - Code cleanup (non-breaking change which makes code smaller or more readable)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: plugins/linux-capture/xcomposite-input.c
Comment (kkartaltepe) [inline plugins/linux-capture/xcomposite-input.c:384]: We can drop the X error handler as well, its not being called in the new egl path.

**Comment by kkartaltepe** [comment]
It looks like i missed an include in my earlier rewrite for the X11 headers which was being covered by the glx headers. Woops.

File: plugins/linux-capture/xcomposite-input.c
Comment (GeorgesStavracas) [reply plugins/linux-capture/xcomposite-input.c]: Good catch!

**Comment by GeorgesStavracas** [comment]
This needs to be tested on a variety of hardware, with special attention to NVIDIA and AMD cards. Intel GPUs should handle this transition just fine.

**Comment by sekalengrengginang** [comment]
tested on obs flatpak with Nvidia GTX 1050ti driver 510.68.02
Xcomposite and XSHM work just fine

file:///var/home/shioriko/Pictures/spectacle/Screenshot_20220515_211034.png

**Comment by denisschultz** [comment]
Tested on AMD Radeon RX 5500 XT, Screen capture and Display capture without any problems.

**Comment by univrsal** [comment]
Seems to work fine for me under Nvidia with proprietary drivers. (Arch linux, 5.17.7-zen1-1-zen, Driver version 515.43.04, GTX 970)

**Comment by tychii93** [comment]
Tested with Flatpak.  Both window and screen capture works with Vega 56, Mesa 22.0.3.  Arch Linux with 5.17.8-zen

File: libobs/obs-nix-platform.h
Comment (norihiro) [inline libobs/obs-nix-platform.h:27]: Removing this item, we loose binary compatibility with plugins. These code use the enumerator.
https://github.com/exeldro/obs-source-dock/blob/134f1f614a4382e68de265230aaddc06cc66c8b9/qt-display.cpp#L163-L178
https://github.com/norihiro/obs-color-monitor/blob/e8359b1560c1ea8dede019e8e59943325b48be28/src/obsgui-helper.hpp#L21-L36

Comment (GeorgesStavracas) [reply libobs/obs-nix-platform.h:27]: This is an excellent point that I didn't consider, thanks! Though now that I'm thinking, OBS 28 will break ABI anyway (Qt5 → Qt6) so I think we're accidentally safe for now.

Comment (norihiro) [reply libobs/obs-nix-platform.h:27]: IIRC, since the package for Ubuntu 20.04 (PPA) will be still compiled with Qt5, isn't that still ABI compatible?

Comment (kkartaltepe) [reply libobs/obs-nix-platform.h:27]: Qt is not part of the ABI, but even if it were we should seek to reduce the pain not use it to add more. We can just add a _UNUSED to the name to signify its no longer used but just deleting it makes it that much harder to detect the ABI break as   a plugin developer.

Comment (GeorgesStavracas) [reply libobs/obs-nix-platform.h:27]: The enum value is still there now, but nothing can set it, and there's a comment instructing to stop using it in code. Is that good now?

File: libobs-opengl/gl-nix.c
Comment (norihiro) [inline libobs-opengl/gl-nix.c:33]:
```suggestion
```
How about just removing the case item `OBS_NIX_PLATFORM_X11_GLX` or put `break` so that it will reach to `assert` at the end?

File: plugins/linux-pipewire/linux-pipewire.c
Comment (norihiro) [inline plugins/linux-pipewire/linux-pipewire.c:48]: I think the entire `switch` statement here can be removed and just call `pipewire_capture_load()`. We can reduce 10 lines.

File: libobs-opengl/gl-nix.c
Comment (GeorgesStavracas) [reply libobs-opengl/gl-nix.c]: Sorry, I had removed this locally but didn't push afterwards... this is gone now.

**Review by kkartaltepe — APPROVED** [review]
I'm still a little apprehensive about cutting out GLX before we have had a release with EGL as the default, but dont let that block merging if you are comfortable with it.

**Comment by trinitronx** [comment]
@Gol-D-Ace :pray: 

[Re-posting this][1] here in hopes that someone with PPA push access will see it.

Can we get the old package version prior to this breaking change pushed back up to the [`obsproject/obs-studio` PPA][2]?  Especially considering LTS releases such as Ubuntu 20.04 Focal.

This was a breaking change for systems with old graphics cards.  Only the latest version is now available in the PPA, and there is no way to pin an old version for such systems with old hardware.

[1]: https://obsproject.com/forum/threads/obs-doesnt-start-any-longer-since-latest-update.159270/#post-587553
[2]: https://launchpad.net/~obsproject/+archive/ubuntu/obs-studio

**Comment by derrod** [comment]
PPA's cannot publish multiple versions at the same time, but you can manually download the old package here: https://launchpad.net/~obsproject/+archive/ubuntu/obs-studio/+packages

**Pull request merged**
